### PR TITLE
Feature branch for Bubble Sorting

### PR DESCRIPTION
**Pull request by shubham14p3**
**Created our first bubble sort project in ruby**
We had two modules as per requirement
#bubble_sort
#bubble_sort_by

Inputs for the same are as follow :
**a.  #bubble_sort**
array = [5,9,7,3,1]
bubble_sort(array)

**b.  #bubble_sort_by**
array = ["hi","hello","hey"]
bubble_sort_by(array) do |left,right|
	right.length - left.length
end

Output for the above were as follow:-

**a.  #bubble_sort**
This is the list that needs to be sorted : [5, 9, 7, 3, 1]
This is the sorted list [1, 3, 5, 7, 9]

**b.  #bubble_sort_by**
This is the list that needs to be sorted : ["hi", "hello", "hey"]
This is the sorted list after sorting : ["hi", "hey", "hello"]

### CODE REVIEW

**Comment by shubham14p3** [comment]
Note:
The Live Page will be available once the code is merged with main

With Love
@shubham14p3 
@rootDEV2990

File: sort_by.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+		while swapped do
+		swapped = false
+		(n - 1).times do |i|
+			if array[i] > array[i + 1]
+			array[i], array[i + 1] = array[i + 1], array[i]
+			swapped = true
+			end
+		end
+	end
+	puts "This is the sorted list #{array}"
```
Comment (kobiyoyo) [inline sort_by.rb]: Let it just output only the array no strings

File: sort_by.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		swapped = 1
+		while swapped < passes do
+			if yield(array[swapped-1],array[swapped]) < 0
+				array[swapped - 1], array[swapped] = array[swapped], array[swapped-1]
+			end
+			swapped += 1
+		end
+		passes -= 1
+	end
+	puts "This is the sorted list after sorting : #{array}"
```
Comment (kobiyoyo) [inline sort_by.rb]: Let it just output only the array no strings

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team   
#### Great Job so far, you are almost there .
### General

- [ ] Enable stickler for this project
- [ ] `bubble_sort_by`
 ` > bubble_sort_by(["hi","hello","hey"]) do |left,right|`
  `>   left.length - right.length`
 ` > end`
 ` => should return ["hi", "hey", "hello"]`
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: sort_by.rb (hunk 1 above)
Comment (shubham14p3) [reply sort_by.rb]: @kobiyoyo Could you just explain it to me why DO i need to do the following , its not mentioned in the criteria that we cant have a string as out put.
I have made the changes but Still want to know what made you say that I need to just print array without strings ?

File: sort_by.rb (hunk 2 above)
Comment (shubham14p3) [reply sort_by.rb]: @kobiyoyo Could you just explain it to me why DO i need to do the following , its not mentioned in the criteria that we cant have a string as out put.
I have made the changes but Still want to know what made you say that I need to just print array without strings ?

**Comment by shubham14p3** [comment]
Closing the Pull Request, Having problem in enabling the Stickler,
Creating an new pull Request for the same.

Also have made the changes told by tse.

**Pull request closed without merging**